### ISSUE TITLE
no way to bing SSL to hostname?

### ISSUE THREAD

**Issue by docgregt** [Azure/azure-cli#9804]
How to bind the SSL to a hostname?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2a3f984d-0d48-e075-7e2f-2da5667b15cf
* Version Independent ID: 98e06d47-0ed4-4d8e-ccf5-d3a78e510539
* Content: [az webapp config ssl](https://docs.microsoft.com/en-us/cli/azure/webapp/config/ssl?view=azure-cli-latest#az-webapp-config-ssl-bind)
* Content Source: [latest/docs-ref-autogen/webapp/config/ssl.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/live/latest/docs-ref-autogen/webapp/config/ssl.yml)
* Service: **app-service-web**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by yugangw-msft** [comment]
The SSL cert you purchased will let you provide the hostname. Once the certificate is ready, you can run `az webapp config ssl upload` to have your webapp configured.
Please reactivate if you have further questions.

**Comment by docgregt** [comment]
I dont think you understood the question or I didnt make it clear. We already have SSL uploaded and bound to some hostnames. The issue with this cli cmd is that it rebinds all hostnames and there is no way to specify a single hostname. What if I dont want to rebind all the hostnames every time I add a new hostname or want different certs with different hostnames?

**Comment by yugangw-msft** [comment]
@docgregt, sorry for closing this prematurely. I am reactivating this and let @panchagnula follow up

**Comment by panchagnula** [comment]
@docgregt currently not something that CLI supports & this is the first request we are receiving for mapping a single certificate to a specific hostname. Need to run this by the feature crew to add support.  will get back to you on this.

**Comment by dennisahlin** [comment]
Workaround:
1. Get the hostname id, for example by listing the hostnames of your webapp like so: `az webapp config hostname list --resource-group $resourceGroupName --webapp-name $webAppName`.
2. Bind the ssl certificate by using ids parameter: `az webapp config ssl bind --certificate-thumbprint $sslThumbprint --ssl-type SNI --ids $hostnameId`.

**Comment by btardif** [comment]
We are reviewing design for SSL/TLS related features.

**Comment by docgregt** [comment]
If you are reviewing it then it would be nice to handle *.domain.com. We have star cert so we are half way there.
Right now for multiclient situation we have to add each domain name 1 at a time. Such as:
a.domain.com
b.domain.com
c.domain.com

etc.

Gregory Turner
Chief Architect
________________________________________
i4 Insight, Inc.

cell: +1.720.335.1744
greg.turner@i4-insight.com<mailto:greg.turner@i4-insight.com>

A member of the Lloyd’s Register Group

**Comment by SQLLord** [comment]
You also need to remember that when it does this it does not do the scm name of the web site when it tries to bind and not everyone can use wild card certs.

**Comment by panchagnula** [comment]
We will update this with details after we review this with the SSL feature crew.

**Comment by panchagnula** [comment]
@btardif something we need to follow up with the SSL feature crew.

**Comment by panchagnula** [comment]
@yutanglin16 lets discuss this some time next week. Thanks!

**Comment by panchagnula** [comment]
Sent email to the feature crew for design meeting so we can start working on implementation when design is approved.

**Comment by simondevolutions** [comment]
I would like to add that this is a critical feature for us too. We have websites with hundreds of bindings and in the current situation, renewing our wildcard certificates might be really painful if we need to update binding one by one in the portal.

**Comment by jtemperv** [comment]
Stumbled on this thread as I was searching for a way to do this. Can anyone confirm that the --ids argument is a way to work around for now?

**Comment by panchagnula** [comment]
@singlessis we should add hostname parameter support to az webapp config ssl bind command https://docs.microsoft.com/en-us/cli/azure/webapp/config/ssl?view=azure-cli-latest#az_webapp_config_ssl_bind

**Comment by panchagnula** [comment]
@Kotasudhakarreddy / @ThejaChoudary got a CRI on this today - when you have cycles lets consider this for the next sprint.

**Comment by Kotasudhakarreddy** [comment]
> @Kotasudhakarreddy / @ThejaChoudary got a CRI on this today - when you have cycles lets consider this for the next sprint.

Sure sisira.

**Comment by Shert** [comment]
I add that it would be nice to have this feature also for the scm part of the webapp, 
cause this part is not covered by the great  workaround suggested by @dennisholmer 
and powershell has this features, for example New-AzWebAppSSLBinding , 
Remove-AzWebAppSSLBinding support the -Name switch and it is possible to work 
on a specific fqdn.

M.

**Comment by navba-MSFT** [comment]
@docgregt Apologies for the late reply. We are looking into this and we will provide an update on this thread once we have more details.

@panchagnula @Kotasudhakarreddy Could you please provide an update on this ? Awaiting your reply.

**Comment by panchagnula** [comment]
no ETA at this point for support on our end. Thanks!

**Comment by navba-MSFT** [comment]
@Kotasudhakarreddy @panchagnula Thanks for the update.
@docgregt Please check the above comments. In the meantime, if you have any further questions on this, please free feel to ask. We would be happy to assist. Awaiting your reply.

**Comment by docgregt** [comment]
> @Kotasudhakarreddy @panchagnula Thanks for the update. @docgregt Please check the above comments. In the meantime, if you have any further questions on this, please free feel to ask. We would be happy to assist. Awaiting your reply.

No further questions. It would be great to get it done.

**Comment by ItsRobbAllen** [comment]
I just found this searching for a solution to this exact problem.

We have a * certificate & we need to add a new hostname and bind just that host to the * cert. I'm trying

> Workaround:
> 
> 1. Get the hostname id, for example by listing the hostnames of your webapp like so: `az webapp config hostname list --resource-group $resourceGroupName --webapp-name $webAppName`.
> 2. Bind the ssl certificate by using ids parameter: `az webapp config ssl bind --certificate-thumbprint $sslThumbprint --ssl-type SNI --ids $hostnameId`.

but the command doesn't seem to work with the id that is returned (which is the URL to the resource).

We have automated systems to build our clients' sites out, but this we have to do manually so it would be very helpful to be able to bind an existing cert to a new hostname.
